pkg/codegen/pcl/binder_schema.go

```diff
@@ -499,10 +499,11 @@ func getDiscriminatedUnionObjectItem(t model.Type) (string, model.Type) {
 // no member if found, (nil, true) returned. If the query is nonsensical, either
 // because no schema is associated with the EnumMember or if the type of value
 // mismatches the type of the schema, (nil, false) is returned.
-func EnumMember(t *model.EnumType, value cty.Value) (*schema.Enum, bool) {
+func EnumMember(t *model.EnumType, value cty.Value) (*schema.Enum, []interface{}, bool) {
+	validMembers := make([]interface{}, 0)
```

Comment:
I'm confused by how `validMembers` works. From the implementation, it looks like we can describe `validMembers` as 

```
A slice of elements of the same type as `value` that are valid values for the
enum `t` that exist before the returned `*schema.Enum`, if any.
```

I'm pretty sure that `validMembers` is only useful if `nil, validMembers, true` is returned. It's a little confusing.

I think valid members could be more easily generated with a separate method on an `*model.EnumType`:

```go
func EnumMemberValues(t *model.EnumType) []interface{} {
  srcBase, ok := GetSchemaForType(t)
  if !ok { return nil }
  src := srcBase.(*schema.EnumType)
  members := make([]interface{}, len(src.Elements))
  for i, el := range src.Elements {
    members[I] = el.Value
  }
  return members
}
```

Reply:
It's only valid if `nil, validMembers, false` is returned, which is the default case for no match. Was just trying to do it in one pass but agree the readability issues are not worth it.